### PR TITLE
CAMEL-21001: camel-jbang Kubernetes - Add Route trait

### PR DESCRIPTION
**Pull request by gansheer**
# Description

* Implement Route trait that is capable of configuring the Openshift Route resource
* Fix parsing of trait profile

# Target

- [X] I checked that the commit is targeting the correct branch (note that Camel 3 uses `camel-3.x`, whereas Camel 4 uses the `main` branch)

# Tracking
- [X] If this is a large change, bug fix, or code improvement, I checked there is a [JIRA issue](https://issues.apache.org/jira/browse/CAMEL) filed for the change (usually before you start working on it).

# Apache Camel coding standards and style

- [X] I checked that each commit in the pull request has a meaningful subject line and body.

- [X] I have run `mvn clean install -DskipTests` locally from root folder and I have committed all auto-generated changes.

### CODE REVIEW

File: dsl/camel-jbang/camel-jbang-plugin-kubernetes/pom.xml [hunk 1]
```diff
@@ -54,6 +54,13 @@
             <version>${kubernetes-client-version}</version>
         </dependency>
 
+        <!-- Kubernetes -->
+        <dependency>
```
Comment (davsclaus) [inline dsl/camel-jbang/camel-jbang-plugin-kubernetes/pom.xml:58]: Why is this JAR added ?

**Review by oscerd — APPROVED** [review]

File: dsl/camel-jbang/camel-jbang-plugin-kubernetes/pom.xml (hunk 1 above)
Comment (davsclaus) [reply dsl/camel-jbang/camel-jbang-plugin-kubernetes/pom.xml:58]: I mean did this JAR bring any code that this plugin needs or was it added for some other reason? Those JARs are a bit big so we should avoid if not needed.

Comment (gansheer) [reply dsl/camel-jbang/camel-jbang-plugin-kubernetes/pom.xml:58]: It brings the Route classes for the Openshift Route ressource, but I see what you means. Let me check if the model jar is enough in our case.

Comment (gansheer) [reply dsl/camel-jbang/camel-jbang-plugin-kubernetes/pom.xml:58]: I replaced the client jar with the model jar.

**Review by davsclaus — APPROVED** [review]

**Pull request merged**